earth_enterprise/src/fusion/autoingest/plugins.src/PacketGen.src

```diff
@@ -247,90 +245,65 @@ void PacketGenAssetVersionImplD::UpdateChildren(
       }
 
       // populate inputs and inputVers & config.insets
-      levInputs.reserve(mergeIndexes.size() * 2 + 1);
-      levInputVers.reserve(mergeIndexes.size() * 2 + 1);
       packetLevelConfig.insets.reserve(mergeIndexes.size() + 1);
-      for (std::vector<uint>::const_iterator idx = mergeIndexes.begin();
-           idx != mergeIndexes.end(); ++idx) {
-        const InsetInfo<ProductAssetVersion> &inset(*extra.insetInfo[*idx]);
-        const PacketGenInfo &genInfo(*extra.genInfo[*idx]);
-        AssetVersion blendVer;
+      packetLevelConfig.insets.resize(mergeIndexes.size());
+      int threadsToUse = MiscConfig::Instance().PackGenThreads;
+      levInputs.resize(mergeIndexes.size() * 2 + 1);
+      levInputVers.resize(mergeIndexes.size() * 2 + 1);
+
+      // Run the following loop in parallel using the amount of threads set in misc.xml.
+      #pragma omp parallel for schedule(static) num_threads(threadsToUse)
+      for (size_t idx = 0; idx < mergeIndexes.size(); ++idx) {
+        const InsetInfo<ProductAssetVersion> &inset(*extra.insetInfo[mergeIndexes[idx]]);
+        const PacketGenInfo &genInfo(*extra.genInfo[mergeIndexes[idx]]);
+        int i = idx*2;
+        levInputs[i] = (inset.combinedrp_).Ref();
+        levInputVers[i] = inset.combinedrp_;
+
         if (!inset.alphaRPFile.empty() &&
             (genInfo.packetLevels.size() > minifySrcLevel)) {
-          blendVer = genInfo.packetLevels[minifySrcLevel];
+          levInputs[i+1] = (genInfo.packetLevels[minifySrcLevel]).Ref();
```

Comment:
Technically there's a difference between this and the old version. The old version worked like this:
```
        if (!inset.alphaRPFile.empty() &&
            (genInfo.packetLevels.size() > minifySrcLevel)) {
          blendVer = genInfo.packetLevels[minifySrcLevel];
        }

        if (!blendVer) {
          levInputs.push_back(inset.combinedrp_->GetRef());
          levInputVers.push_back(inset.combinedrp_);
        }
```
This change shortens it to
```
        if (!inset.alphaRPFile.empty() &&
            (genInfo.packetLevels.size() > minifySrcLevel)) {
          levInputs[i+1] = (genInfo.packetLevels[minifySrcLevel]).Ref();
          levInputVers[i+1] = genInfo.packetLevels[minifySrcLevel];
        }
```
There's a chance that `blendVer` might still evaluate to false even after `blendVer = genInfo.packetLevels[minifySrcLevel];`. In that case `levInputs` and `levInputVers` would never be set, so the outcome of the two versions of the code is different. I'm not sure if it matters in this case, but it's safest to keep the behavior the same.